modern/src/maki-interpreter/virtualMachine.js

```diff
@@ -45,13 +49,13 @@ export function run({
     // const logger = log ? printCommand : logger;
     // TODO: Handle disposing of this.
     // TODO: Handle passing in variables.
-    variable.hook(method.name, (...args) => {
+    variable.hook(method.name, async (...args) => {
       // Interpret is a generator that yields before each command is exectued.
       // `handler` is reponsible for `.next()`ing until the program execution is
       // complete (the generator is "done"). In production this is done
       // synchronously. In the debugger, if execution is paused, it's done
       // async.
-      debugHandler(interpret(commandOffset, program, args.reverse()));
+      await debugHandler(interpret(commandOffset, program, args.reverse()));
```

Comment:
Do we actually need to await here? I think it would be fine to return form this function without waiting for the promise to resolve.

Reply:
whoops. tested it and we do not need it, thanks!